Task2/src/validation/BusTicketValidator.java

```diff
@@ -0,0 +1,131 @@
+        try {
+            boolean valid = true;
+            for (int i = 0; i < tickets.length(); i++) {
+                ticket = tickets.getJSONObject(i);
+
+                valid = validateClass() && valid;
+                valid = validateType() && valid;
+                valid = validateDate() && valid;
+                valid = validatePrice() && valid;
+
+                if (valid) {
+                    totalValid++;
+                }
+                valid = true;
+            }
+            printStats(tickets.length());
+        } catch (JSONException e) {
+            System.out.println("Incorrect JSON format!");
+            System.out.println(e.getMessage());
+        } catch (DateTimeParseException e) {
+            System.out.println("[startDate] format is incorrect!");
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private boolean validateClass() {
+        boolean valid = true;
+        if (ticket.isNull("ticketClass")) {
+            classViolated++;
+            valid = false;
+        }
+        return valid;
+    }
+
+    private boolean validateType() {
+        boolean valid = true;
+
+        if (ticket.isNull("ticketType")) ticketType = "";
+        else ticketType = String.valueOf(ticket.get("ticketType"));
+
+        try {
+            TicketType.valueOf(ticketType);
+        } catch (IllegalArgumentException e) {
+            typeViolated++;
+            valid = false;
+        }
+
+        return valid;
+    }
+
+    private boolean validateDate() throws DateTimeParseException {
+        LocalDate startDate;
+        boolean valid = true;
+        boolean dateRequiringType = ticketType.equals(TicketType.DAY.name())
+                || ticketType.equals(TicketType.WEEK.name()) || ticketType.equals(TicketType.YEAR.name());
+
+        if (dateRequiringType) {
+            if ((ticket.isNull("startDate") || ticket.getString("startDate").equals(""))) {
+                valid = false;
+                dateViolated++;
+            } else {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                startDate = LocalDate.parse(ticket.getString("startDate"), formatter);
+                if (startDate.isAfter(LocalDate.now())) {
+                    valid = false;
+                    dateViolated++;
+                }
+            }
+        } else {
+            if (!ticket.isNull("startDate")) {
+                valid = false;
+                dateViolated++;
+            }
+        }
+        return valid;
+    }
+
+    private boolean validatePrice() {
+        boolean valid = true;
+        if (ticket.isNull("price") || ticket.getFloat("price") == 0) {
+            valid = false;
+            priceViolated++;
+        } else if (ticket.getFloat("price") % 2 != 0) {
+            valid = false;
+            priceViolated++;
+        }
+        return valid;
+    }
+
+    private void printStats(int total) {
```

Comment:
From method name not understandable what will be the params and why do we need only total here :)
Better to put some Statistic object as param and use it inside